.github/workflows/ci.yml

```diff
@@ -66,14 +66,14 @@ jobs:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-session-token: ${{ secrets.AWS_SESSION_TOKEN }}
-          role-to-assume: arn:aws:iam::037018655140:role/brightspace-ui-core.d2l.dev-githubactions-access
+          role-to-assume: arn:aws:iam::022062736489:role/r+BrightspaceUI+core+repo
           role-duration-seconds: 3600
           aws-region: ca-central-1
 
       - name: Publish
         uses: BrightspaceUI/actions/publish-to-s3@main
         with:
-          bucket-path: s3://brightspace-ui-core.d2l.dev/branches/pr-${{ github.event.number }}
+          bucket-path: s3://d2l.dev/pr-previews/brightspace-ui-core/pr-${{ github.event.number }}
```

Comment:
I think I pictured this as:
`bucket-path: s3://d2l.dev/pr-previews/BrightspaceUI/core/pr-${{ github.event.number }}`

But I guess it doesn't _really_ matter, as long as there are no collisions in what people choose 🤔

Reply:
If we did do `org/repo`, we could actually add a special thing to `repo_settings` so someone could do:
```
pr_previews: true
```
and under the hood it would setup the `d2ldev` piece.  But that doesn't work with the "main" site piece.

Just spit-balling here - what are your thoughts on this:
- All previews go into `d2l.dev/pr-previews`
  - `pr-previews/main/<org>/<repo>/` for the main site that doesn't expire
  - `pr-previews/prs/<org>/<repo>/pr-<#>` for the PR sites
- The expiry prefix is changed to `pr-previews/prs`
- For now, someone sets this in `repo-settings` (need to confirm this doesn't blow up):
  ```
  d2ldev:
    s3:
      - subdomain: pr-previews
        path: main/<org>/<repo>
      - subdomain: pr-previews
        path: prs/<org>/<repo>
  ```
  - But we eventually switch this to `pr_previews: true` and set it for them
- If someone wants a fancy path like `brightspace-ui-core.d2l.dev`, they can set it up in `d2l.dev`.  But they don't actually _have_ to set up anything in `d2l.dev` if they're fine with `https://pr-previews.d2l.dev/prs/BrightspaceUI/core/pr-${{ github.event.number }}/` and `https://pr-previews.d2l.dev/main/BrightspaceUI/core/`

Reply:
I'm open to that, except that I think the phrase "PR preview" doesn't apply any more to the "main" static builds.

What about calling them all `repo-demos` or `code-demos`?

Reply:
Yeah good with either of those!  Or even `demos` could work, if we want to try to make the default urls as easy to remember as possible (maybe not useful with url autocomplete lol)

Reply:
Okay I might be back to being against this, mostly because of how the URLs would work:

- Suppose the generic prefix was `demos` (I also like the prefix `static`)
- So we have PR previews at e.g. https://demos.d2l.dev/BrightspaceUI/core/prs/pr-1234/, which sounds fine
    - or maybe we could remove the `prs/` piece: https://demos.d2l.dev/BrightspaceUI/core/pr-1234/
- But then the main demo would either be at:
    - https://demos.d2l.dev/BrightspaceUI/core/main/ (ugly), or
    - https://demos.d2l.dev/BrightspaceUI/core/ (which is clean but means the repo is _forced_ to avoid the special folder name `prs` or `pr-.+`)

I don't like that folder restriction, but maybe it's fine...

Reply:
Wouldn't the `pr` and `main` need to be at the beginning to be able to set the retention properly, since we can't use wildcards in the prefix?  Like https://demos.d2l.dev/main/BrightspaceUI/core/ (or I think we could get away with https://demos.d2l.dev/BrightspaceUI/core/) and https://demos.d2l.dev/prs/BrightspaceUI/core/

I guess we could map it in some way with the routes, but this way maybe alleviates your concern cause there's no folder name they need to avoid. They publish their main site to `d2ldev/demos/<org>/<repo>` and their PR sites to `d2ldev/demos/prs/<org>/<repo>/<whatever-they-want-really>-<pr_num>`.  And the lifecycle prefix is "/demos/prs/"?

Reply:
Yeah the prefix would need to be there in the bucket, but like you said the routing doesn't need to match the bucket.

I like keeping the main demo link clean as https://demos.d2l.dev/BrightspaceUI/core/. You're right that sticking the `prs` prefix into the URL works, though it's a bit buried there...

We could also just do a completely different route and serve the `prs/` bucket up at https://pr-previews.d2l.dev/BrightspaceUI/core/. That might be an anti-pattern though, `d2l.dev` is generally set up so the bucket prefix matches the subdomain.

So then - why not just two separate bucket prefixes? Keep `pr-previews` as it currently is, and just add one for `demos`? Is there any benefit to trying to merge them into one, apart from their content being similar? Keeping `pr-previews` clearly separate might also make the expiration rule more obvious.

Reply:
Alright I thought on it a bit, I can get behind https://demos.d2l.dev/prs/BrightspaceUI/core/ - the URL for these aren't too important anyway, since you're only going to be visiting them as links from PRs. And for that same reason, having a whole subdomain dedicated to them isn't too useful.

Any other thoughts before I go configure that in d2l.dev?

Reply:
I'm good with either of your last two comments - going with https://demos.d2l.dev/prs/BrightspaceUI/core/, or just making two bucket prefixes to keep things clear, since we could still wrap that in `repo-settings` behind a single command.  I was trying to think of two buckets prefixes that worked nicely together lol... `demo` and `demo-prs`? `pr-preview` and `repo-preview`? `preview` and `pr-preview`?